File: ReadyTasks/FSharp/ProjectEuler/Task_5/Task_5/Program.fs
open System

let isPrime numb =
    Seq.initInfinite (fun x -> (int64)x + 2L) |> Seq.takeWhile (fun x -> x * x <= numb) |> Seq.forall (fun x -> numb % x > 0L)

let rec gcd a b =
    if (a % b = 0L) then
        b
    else
        gcd b (a % b)

let getMaxProduct maxVal =
    [2L..maxVal] |> List.fold (fun acc x -> acc * x / (gcd acc x)) 1L

[<EntryPoint>]
let main argv =
    let result = getMaxProduct 20L
    printfn "%d" result
    0 // return an integer exit code

File: ReadyTasks/CSharp/ProjectEuler/Task_5/Task_5/Program.cs
using System;

namespace Task_5
{
    class Program
    {
        static long GCD(long a, long b)
        {
            return (b == 0) ? a : GCD(b, a % b);
        }

        static long MaxVal(int max)
        {
            long result = 1;
            for (int i = 2; i <= max; i++)
            {
                long gcd = GCD(result, i);
                result *= i;
                result /= gcd;
            }
            return result;
        }
        static void Main(string[] args)
        {
            Console.WriteLine(MaxVal(20));
        }
    }
}
